packages/lib/passport-interface/src/Credential.ts

```diff
@@ -0,0 +1,132 @@
+import { EdDSAPCDClaim } from "@pcd/eddsa-pcd";
+import { EmailPCD, EmailPCDClaim, EmailPCDPackage } from "@pcd/email-pcd";
+import { SerializedPCD } from "@pcd/pcd-types";
+import {
+  SemaphoreSignaturePCD,
+  SemaphoreSignaturePCDClaim,
+  SemaphoreSignaturePCDPackage
+} from "@pcd/semaphore-signature-pcd";
+import { ONE_HOUR_MS, ONE_MINUTE_MS } from "@pcd/util";
+
+// Timestamps last for one hour and 20 minutes
+// Compare to CACHE_TTL in CredentialManager.ts, which is one hour, meaning
+// that client-side cached credentials will be refreshed before they expire
+const TIMESTAMP_MAX_AGE = ONE_HOUR_MS + 20 * ONE_MINUTE_MS;
+
+// To avoid writing `SerializedPCD<SemaphoreSignaturePCD>`, and also to make
+// the `Credential` type a bit less tightly-bound to the implementation details
+// of serialiazation and Semaphore signatures, we use this type.
+export type Credential = SerializedPCD<SemaphoreSignaturePCD>;
+
+/*
+ * The payload encoded in the message of the SemaphoreSignaturePCD passed
+ * as a credential to feeds.
+ */
+export interface CredentialPayload {
+  // The only type of PCD that can appear here is EmailPCD.
+  pcd?: SerializedPCD<EmailPCD>;
+  timestamp: number;
+}
+
+/**
+ * The result of successfully verifying a credential.
+ *
+ * We do not need to return whole PCDs here, because the proofs have been
+ * verified, and since we expect to cache these in memory, we can avoid wasting
+ * memory on caching large PCD objects.
+ */
```

Comment:
a little more context on what exactly it means to be `verified` could be useful here - e.g. that the key that was used to sign the credential matches the zupass public key, etc.

Reply:
Added